src/raftstore/store/store.rs

```diff
@@ -1789,6 +1790,70 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         };
     }
 
+    fn on_compaction_finished(&mut self, event: CompactedEvent) {
+        // If size declining is trivial, skip.
+        let total_bytes_declined = if event.total_input_bytes > event.total_output_bytes {
+            event.total_input_bytes - event.total_output_bytes
+        } else {
+            0
+        };
+        if total_bytes_declined < self.cfg.region_split_check_diff.0 ||
+            total_bytes_declined * 10 < event.total_input_bytes
+        {
+            return;
+        }
+
+        let output_level_str = event.output_level.to_string();
+        COMPACTION_DECLINED_BYTES
+            .with_label_values(&[&output_level_str])
+            .observe(total_bytes_declined as f64);
+
+        // Calculate influenced regions.
+        let mut influenced_regions = vec![];
+        for (end_key, region_id) in self.region_ranges
+            .range((Included(event.start_key), Included(event.end_key)))
+        {
+            influenced_regions.push((region_id, end_key.clone()));
+        }
+        // Compaction may happens between the gap of two valid regions.
+        if influenced_regions.is_empty() {
+            return;
+        }
+
+        let mut region_declined_bytes = vec![];
+        let mut last_end_key = data_key(b"");
+        for (region_id, end_key) in influenced_regions.drain(..) {
+            let mut old_size = 0;
+            for prop in &event.input_props {
+                old_size += prop.get_approximate_size_in_range(&last_end_key, &end_key);
+            }
+            let mut new_size = 0;
+            for prop in &event.output_props {
+                new_size += prop.get_approximate_size_in_range(&last_end_key, &end_key);
+            }
+            last_end_key = end_key.clone();
+
+            if old_size > new_size &&
+                old_size - new_size > self.cfg.region_split_check_diff.0 / 16
```

Comment:
This check seems not necessary, any reason to add this check?

Reply:
For filtering some regions.